File: Python/515. Find Largest Value in Each Tree Row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def largestValues(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        ans = list()
        def dfs(node, depth):
            if node is None:
                return None
            if len(ans) <= depth:
                ans.append(node.val)
            elif ans[depth] < node.val:
                ans[depth] = node.val
            dfs(node.left, depth+1)
            dfs(node.right, depth+1)
        dfs(root, 0)
        return ans

File: C++/515. Find Largest Value in Each Tree Row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* node, int depth, vector<int> &ans)
    {
        if(node == nullptr)
            return ;
        if(ans.size()<=depth)
            ans.push_back(node->val);
        else if(ans[depth] < node->val)
            ans[depth] = node->val;
        dfs(node->left, depth+1, ans);
        dfs(node->right, depth+1, ans);
    }
    vector<int> largestValues(TreeNode* root) {
        vector<int> ans;
        dfs(root, 0, ans);
        return ans;
    }
};